samples/templates/express-direct-auth/web-server/routes/recover-password.js

```diff
@@ -1,9 +1,137 @@
 const express = require('express');
+const { getAuthClient } = require('../utils');
 
 const router = express.Router();
 
+const authenticators = ['email']; // ordered authenticators
+// const authenticators = ['phone']; // ordered authenticators
+
+const handleAuthTransaction = (req, res, { authClient, authTransaction }) => {
+  const {  
+    data: { 
+      nextStep,
+      interactionHandle,
+      tokens,
+    }
+  } = authTransaction;
+
+  const next = () => {
+    if (nextStep.name === 'challenge-authenticator') {
+      res.redirect(`/recover-password/challenge-${nextStep.type}-authenticator`);
+    } else if (nextStep.name === 'reset-authenticator') {
+      res.redirect('/recover-password/reset');
+    } else {
+      throw { errorCauses: ['Unable to handle next step.'] };
```

Comment:
we should always throw objects which are instanceof `Error`. Should we add a new custom error type?

Reply:
This is error in sample, just to serve the template.

Reply:
It's cool to collect the errors in an array and display them, but we should throw an instance of the `Error` object so that it has stack trace information. Can you do `throw new Error('Unable to handle next step.')` and then, in the catch block, add `err.message` to the errors array?